modules/sor/sorV2/sorV2.service.ts

```diff
@@ -0,0 +1,243 @@
+                            balance: t.dynamicData?.balance as unknown as HumanAmount
+                    }
+                }),
+            };
+            if(["Weighted", "Investment", "LiquidityBootstrapping"].includes(rawPool.poolType)) {
+                rawPool = {
+                    ...rawPool,
+                    tokens: rawPool.tokens.map((t, i) => { return {...t, weight: prismaPool.tokens[i].dynamicData?.weight} }),
+                } as RawWeightedPool;
+            }
+            if(rawPool.poolType === "Stable") {
+                rawPool = {
+                    ...rawPool,
+                    amp: prismaPool.stableDynamicData?.amp,
+                } as RawStablePool;
+            }
+            if(["MetaStable", "ComposableStable"].includes(rawPool.poolType)) {
+                rawPool = {
+                    ...rawPool,
+                    amp: prismaPool.stableDynamicData?.amp.split('.')[0], // Taken b-sdk onChainPoolDataEnricher.ts
+                    tokens: rawPool.tokens.map((t, i) => { return {...t, priceRate: prismaPool.tokens[i].dynamicData?.priceRate} }),
+                } as RawMetaStablePool;
+            }
+            if(rawPool.poolType === "Linear") {
+                rawPool = {
+                    ...rawPool,
+                    mainIndex: prismaPool.linearData?.mainIndex, 
+                    wrappedIndex: prismaPool.linearData?.wrappedIndex,
+                    lowerTarget: prismaPool.linearDynamicData?.lowerTarget,
+                    upperTarget: prismaPool.linearDynamicData?.upperTarget,
+                    tokens: rawPool.tokens.map((t, i) => { return {...t, priceRate: prismaPool.tokens[i].dynamicData?.priceRate} }),
+                } as RawLinearPool;
+            }
+            return rawPool;
+        });
+    }
+
+    /**
+     * Map b-sdk RawPools to BasePools.
+     * @param pools
+     * @returns 
+     */
+    private mapToBasePools(pools: RawPool[]): BasePool[] {
+        const chainId = networkContext.chainId as unknown as ChainId;
+        return sorParseRawPools(chainId, pools);
+    }
+
+    /**
+     * Map Prisma pool type to b-sdk Raw pool type.
+     * @param type 
+     * @returns 
+     */
+    private mapRawPoolType(type: PrismaPoolType): SupportedRawPoolTypes | string {
+        // From b-sdk:
+        // - type LinearPoolType = `${string}Linear`;
+        // - LinearPoolType | 'Weighted' | 'Investment' | 'LiquidityBootstrapping' | 'Stable' | 'MetaStable' | 'ComposableStable' | 'StablePhantom' | 'Element';
+        switch(type) {
+            case PrismaPoolType.WEIGHTED:
+                return 'Weighted';
+            case PrismaPoolType.INVESTMENT:
+                return 'Investment';
+            case PrismaPoolType.LIQUIDITY_BOOTSTRAPPING:
+                return 'LiquidityBootstrapping';
+            case PrismaPoolType.STABLE:
+                return 'Stable'; // TODO - Is there a ComposableStable/Stable differentiation in Prisma?
```

Comment:
I couldn't see any differentiation between Stable/ComposableStable pool types. Is it possible to determine this somehow?

Reply:
Composablestables are phantomstables in prisma. There is an additional check isComposableStable()